packages/react/menu/src/Menu.tsx

```diff
@@ -438,16 +437,49 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
     ...contentProps
   } = props;
   const context = useMenuContext(CONTENT_NAME);
-  const typeaheadProps = useMenuTypeahead();
   const { getItems } = useCollection();
   const [currentItemId, setCurrentItemId] = React.useState<string | null>(null);
   const contentRef = React.useRef<HTMLDivElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, contentRef, context.onContentChange);
   const isPointerDownOutsideRef = React.useRef(false);
+  const timerRef = React.useRef(0);
+  const searchRef = React.useRef('');
+  const [isSearching, setIsSearching] = React.useState(false);
 
   const PortalWrapper = portalled ? Portal : React.Fragment;
   const ScrollLockWrapper = disableOutsideScroll ? RemoveScroll : React.Fragment;
 
+  const handleTypeaheadSearch = (key: string) => {
+    const search = searchRef.current + key;
+    const items = getItems().filter((item) => !item.disabled);
+    const currentItem = document.activeElement;
+    const currentMatch = items.find((item) => item.ref.current === currentItem)?.textValue;
+    const values = items.map((item) => item.textValue);
+    const nextMatch = getNextMatch(values, search, currentMatch);
+    const newItem = items.find((item) => item.textValue === nextMatch)?.ref.current;
+
+    (function updateSearch(value: string) {
+      searchRef.current = value;
+      setIsSearching(value.length > 0);
+      window.clearTimeout(timerRef.current);
+      if (searchRef.current !== '') {
+        timerRef.current = window.setTimeout(() => updateSearch(''), 1000);
+      }
```

Comment:
Looks like we can use `value.length > 0` **or** `value !== ''` for `setIsSearching` and the timeout guard so they're consistent. I don't mind which one but having the two different approaches made me wonder if there was a difference between the two some how... Is there?

```suggestion
      const isSearching = value !== '';
      setIsSearching(isSearching);
      searchRef.current = value;
      window.clearTimeout(timerRef.current);
      if (isSearching) timerRef.current = window.setTimeout(() => updateSearch(''), 100);

Reply:
I had it this way but am thinking of introducing a difference in the morning so wanted to remind myself! I also need to remember the single line if statements (I personally don't like them so it's a hard habit for me to break haha)